unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,378 @@
+
+
+def scroll_wrapper(func, results_key="results"):
+    def wrapper(*args, **kwargs):
+        """Wraps a function to obtain scroll functionality.
+        Function needs to be able to accept 'start' and 'limit' arguments."""
+        if "number_of_items_to_fetch" in kwargs:
+            number_of_items_to_fetch = kwargs["number_of_items_to_fetch"]
+            del kwargs["number_of_items_to_fetch"]
+        else:
+            number_of_items_to_fetch = 100
+
+        kwargs["limit"] = min(100, number_of_items_to_fetch)
+        kwargs["start"] = 0 if "start" not in kwargs else kwargs["start"]
+
+        all_results = []
+        num_iterations = math.ceil(number_of_items_to_fetch / kwargs["limit"])
+
+        for _ in range(num_iterations):
+            response = func(*args, **kwargs)
+            if type(response) is list:
+                all_results += func(*args, **kwargs)
+            elif type(response) is dict:
+                if results_key not in response:
+                    raise KeyError(
+                        "Response object has no known keys to \
+                                   access the results, such as 'results' or 'values'.",
+                    )
+                all_results += func(*args, **kwargs)[results_key]
+            kwargs["start"] += kwargs["limit"]
+
+        return all_results[:number_of_items_to_fetch]
+
+    return wrapper
+
+
+@dataclass
+class JiraIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing).
+
+    Current implementation creates a Jira connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleJiraConfig
+    file_meta: JiraFileMeta
+
+    def grouping_folder_name(self):
+        if self.file_meta.board_id:
+            return self.file_meta.board_id
+        else:
+            return self.file_meta.project_id
+
+    @property
+    def filename(self):
+        download_file = f"{self.file_meta.issue_id}.txt"
+
+        return (
+            Path(self.standard_config.download_dir) / self.grouping_folder_name() / download_file
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        """Create output file path."""
+        output_file = f"{self.file_meta.issue_id}.json"
+
+        return (
+            Path(self.standard_config.output_dir) / self.grouping_folder_name() / output_file
+        ).resolve()
+
+    @requires_dependencies(["atlassian"])
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        from atlassian import Jira
+
+        logger.debug(f"Fetching {self} - PID: {os.getpid()}")
+
+        # TODO: instead of having a separate connection object for each doc,
+        # have a separate connection object for each process
+        jira = Jira(
+            self.config.url,
+            username=self.config.user_email,
+            password=self.config.api_token,
+        )
```

Comment:
we can do this now! this can be done by defining a session handle creator. here is the PR where it was added: https://github.com/Unstructured-IO/unstructured/pull/1058. in it, look at the changes needed for Google Drive. largely: there are some mixins you'll need to add, you create a JiraSessionHandle class, and you'll define `create_session_handle` on the Config.

Reply:
This is great 🚀🚀 - I've added it now